hta/trace_analysis.py

```diff
@@ -504,3 +505,61 @@ def get_cupti_counter_data_with_operators(
                 for individual performance counters.
         """
         return CuptiCounterAnalysis.get_counter_data_with_operators(self.t, ranks)
+
+    def critical_path_analysis_for_rank(
+        self,
+        rank: int,
+        annotation: str,
+        instance_id: Optional[int],
+    ) -> List[pd.DataFrame]:
+        r"""
+        Perform critical path analysis for trace events within a rank.
+        We further reduce the region of interest by selecting
+        a trace annotation and instance id. This will
+        limit the analysis to events within the time range of that annoation.
+        For example, you can use this to limit the analysis to one iteration
+        by passing annotation='ProfilerStep500'.
```

Comment:
It might be useful to add that we should not use the first step as some of the corresponding events may be missing.